### DISCUSSION TITLE
Q&A Setup in TE3

### DISCUSSION THREAD

**Discussion by michael-mertens** [TabularEditor/TabularEditor3#1246] [🙋♀️TE3 Q&A]
Hello, 

I'm currently creating models purely with TE3 and publish to PBI Premium via XMLA endpoint. I have never missed Power BI Desktop.
But now I need to configure the Q&A setup, which is at least not supported by a user interface in TE3. I'm trying to understand the consequences for my work and what kind of workarounds there are.
Like
- switch to PBI Desktop and open TE3 parallelly (I don't know how this is called - i did this years ago with TE2)
- create the setup in a dummy PBI Desktop project and bring it somehow into TE3
- use TE3 like before, export to a PBI file/project (don't know if possible), then do the Q&A setup and publish from there

Any suggestions on that?

Regards
Michael

**Comment by otykier** [comment]
By Q&A setup, did you mean configuring [synonyms](https://learn.microsoft.com/en-us/power-bi/natural-language/q-and-a-best-practices#add-synonyms-to-tables-and-columns)? As far as I know, synonyms are the only type of metadata associated with Q&A, which is defined and stored in the model metadata, so it can be accessed by Tabular Editor. Unfortunately, we don’t have any UI for it in Tabular Editor, since the synonym metadata is not documented anywhere, and Microsoft did not provide an API for it either.

You can find the synonym metadata as a JSON string in the LinguisticMetadata property of the Translations/Culture object in the model.

I think all 3 options should be viable, but option 2 is probably the most tricky, since you’d need to have two identical models before you can copy synonyms from one to the other.

Hope this helps.

**Reply by michael-mertens** [reply]
By Q&A setup, I mean "Q&A setup" ;-) This is the name of the button in Power BI Desktop. And it is actually more than synonyms. There are "Synonyms", "Relationships", "Teach Q&A", "Review questions" and "Suggest questions". The initial linguisticMetadata property of a small pbi project is about 1200 lines before I added anything. "Review questions" is a UI which allows to read questions the users provided for review in the PBI Service. I find this very special because with this functionality PBI Desktop is much more than an editor for json/tmdl files. And the Copilot integration will probably add much more functionality which goes beyond editing a project file.
But back to LinguisticMetadata. I'm not aware of people providing support for the creation of this information (while there are many theme editors). I haven't finished analyzing it, but my gut feeling is, that I can't maintain it manually or semi-automatically (somehow scripted), so that I can continue to work with TE like before. 
I tried to get some more people to comment on this topic, but I don't have enough attention / followers: https://www.linkedin.com/pulse/how-work-te3-2024-michael-mertens-mdgic/?trackingId=%2FWc9V3%2B5Jfij%2BUk58PA5BA%3D%3D
If you think that it is a useful discussion, you might want to comment there. 
My current route is to think about switching to the parallel use of Desktop and TE, which I would consider as a downgrade.
